redis.c

```diff
@@ -2021,32 +2021,25 @@ PHP_METHOD(Redis, discard)
     RETURN_FALSE;
 }
 
-/* redis_sock_read_multibulk_multi_reply */
-PHP_REDIS_API int redis_sock_read_multibulk_multi_reply(INTERNAL_FUNCTION_PARAMETERS,
+PHP_REDIS_API int
+redis_sock_read_multibulk_multi_reply(INTERNAL_FUNCTION_PARAMETERS,
                                       RedisSock *redis_sock)
 {
 
     char inbuf[4096];
     int numElems;
     size_t len;
 
-    if (redis_sock_gets(redis_sock, inbuf, sizeof(inbuf) - 1, &len) < 0) {
-        return - 1;
-    }
-
-    /* number of responses */
-    numElems = atoi(inbuf+1);
-
-    if(numElems < 0) {
-        return -1;
+    if (redis_sock_gets(redis_sock, inbuf, sizeof(inbuf) - 1, &len) < 0 ||
+        *inbuf != TYPE_MULTIBULK || atoi(inbuf + 1) < 0
+    ) {
+        return FAILURE;
     }
 
     array_init(return_value);
 
-    redis_sock_read_multibulk_multi_reply_loop(INTERNAL_FUNCTION_PARAM_PASSTHRU,
-                    redis_sock, return_value, numElems);
-
-    return 0;
+    return redis_sock_read_multibulk_multi_reply_loop(INTERNAL_FUNCTION_PARAM_PASSTHRU,
```

Comment:
I think we're passing `return_value` here twice, since `INTERNAL_FUNCTION_PARAM_PASSTHRU` expands to `execute_data, return_value`